### DISCUSSION TITLE
Paged view mode in Reader

### DISCUSSION THREAD

**Discussion by zzggbb** [spacecowboy/Feeder#175] [Ideas]
I have an E-Ink device, so scrolling is a pretty slow+annoying way to read long articles. 
It would be nice if there was a "paged" view mode:
* text from an article is broken up into pages
* buttons to go to the previous/next page

**Comment by spacecowboy** [comment]
would be a nice option but I have no idea how to implement it, nor would I use it myself.

So, help needed.

**Comment by spacecowboy** [comment]
@zzggbb Does your E-ink device have volume buttons by any chance? Or some other hardware buttons that could be used for scrolling?

**Reply by zzggbb** [reply]
I have a Supernote A5X, it only has a power button. No other buttons. I think some very high-end e-ink devices have volume buttons, but most don't. If you do try to implement this (I'd be very appreciative, but no pressure) I think the best way is with swipe gestures, or tapping near the left/right edges of the screen.

**Reply by spacecowboy** [reply]
Splitting it into pages is out of the question at the moment (articles aren't just text. They are also images, videos, tables etc so really hard to split)

But I have implemented code to scroll one screen up or down with a click, only question is where to put the functionality.

**Comment by tent4kel** [comment]
Thanks for looking into this! Scrolling by page does make a huge difference for e-ink devices. No other rss app is offering that functionality. 

I tested your paged modifications on my boox page. Very promising! Two things:

If I interpreted your code correctly, it is supposed to work in the feed list, but it didn't for me. 

Secondly there is a offset when flipping through the pages. That's a bit of an issue because the eye has to find the line where you left off. 

It seems consistent, so I guess it's that 0.8 modifier in the code. I will try to optimize it for my device by hand. Maybe an option to set it or to define the scroll distance in absolute PX would be a easy to implement solution. 

https://github.com/spacecowboy/Feeder/assets/48975759/cb830fc2-b656-44b7-b605-a2a8e6536e80

**Reply by tent4kel** [reply]
Follow-up question:  scrollHeightPx is not honoring the state of the navbar, right?

**Reply by spacecowboy** [reply]
No, the navBar is not included I think. I left that for later just to get a feel for it.

Thanks for testing it!

**Reply by spacecowboy** [reply]
Oh, so the buttons probably onl work in the *list*. I see in the video you are seeing the grid view. I suspect that I didn't implement it there.
